app/controllers/transfer_packages_controller.rb

```diff
@@ -38,8 +38,28 @@ def show
       .find(params[:id])
 
     @transfer_package = TransferPackagePresenter.new(transfer_package, @navigation_context)
+
     @view_helper = view_helper({ save_back_path: true })
     @can_update = false
+    @can_confirm = !transfer_package.confirmed? && @transfer_package.receiver? && authorize_resource(confirmation_resource(transfer_package), UPDATE_BOX)
+  end
+
+  def confirm
+    transfer_package = TransferPackage
+      .with_receiver(@navigation_context.institution)
+      .find(params[:id])
+    @transfer_package = TransferPackagePresenter.new(transfer_package, @navigation_context)
+
+    return unless authorize_resource(confirmation_resource(transfer_package), UPDATE_BOX)
```

Comment:
The `#confirmation_resource` method takes a Box, and will pass `box.id` to the policy framework, but you're passing a TransferPackage instead? Same above for `@can_confirm`.

I think you mean to use `box = transfer_package.boxes.take` instead?

Reply:
Good catch. I believe the policy framework doesn't actually validate the actual resource ID, otherwise this would've likely failed at some point.

Reply:
I believe it tries to search a permission for `box:id` with a fallback to a broader `institution:id` permission, so it works because I don't think there are actual such fine grained permissions (but there could be).